### ISSUE TITLE
Error in Ripple when navigating

### ISSUE THREAD

**Issue by alecchisi** [jamesmfriedman/rmwc#175]
## Bugs

Follow the template below to ensure the quickest and most accurate response to your issue.

### What RMWC Version are you using?

rmwc 1.4.1

### What browser(s) and React Version is this bug affecting?

> Please include the browser version. A user-agent string is also quite helpful.

### What build system are you using?

React with webpack, create-react-app

When I use List items in my components (I got two container components that use lists)
I got this error from mdc.ripple.js: 

```
Uncaught TypeError: Cannot read property 'getBoundingClientRect' of null
    at Object.computeBoundingRect (mdc.ripple.js:904)
    at MDCRippleFoundation.rmBoundedActivationClasses_ (mdc.ripple.js:1507)
    at MDCRippleFoundation.runDeactivationUXLogicIfReady_ (mdc.ripple.js:1490)
    at MDCRippleFoundation._this.activationTimerCallback_ (mdc.ripple.js:1154)
    at mdc.ripple.js:1432
```

It seems related with the selection of list item.

**Comment by jamesmfriedman** [comment]
@alecchisi I'm not able to reproduce this bug. Can you make a quick repro case?

**Comment by alecchisi** [comment]
Here I am, this is my component: 
```
import React from 'react';


//RMWC components
import { List, ListDivider } from 'rmwc/List';

//Auxiliary component
import Aux from '../../hoc/Auxiliary/Auxiliary'

import ContactsItem from './ContactsItem/ContactsItem';

const contacts = (props) => {

    let users = props.userList;

    //console.log(users);
    return (

            <List avatarList>
                {
                    users.map( user => {
                        console.log("USER ID: "+user.id)
                        return (
                            <Aux key={user.id}>
                            <ContactsItem
                                clicked={() => props.contactSelected (user.id)}
                                username={user.name}
                                avatar={user.avatar}
                            />
                                <ListDivider/>
                            </Aux>
                        )
                    })
                }
            </List>

    );
};

export default contacts;
```

ContactsItem component: 

```
import React from 'react';
import {
    ListItem,
    ListItemGraphic,
    ListItemText

} from 'rmwc/List'

const contactsItem = (props) => {
    return (
        <ListItem
        onClick={props.clicked}
        >
            <ListItemGraphic>
                {props.avatar}
            </ListItemGraphic>
            <ListItemText>
                {props.username}
            </ListItemText>
        </ListItem>
    );
};

export default contactsItem;
```

**Comment by jamesmfriedman** [comment]
Also, I don't know what version of React you are on, but as of 16.2 you can use React.Fragment which does the same thing as your Aux component. Either way, you might want to rename your Aux component to Fragment to give your self a clean upgrade path.

**Comment by alecchisi** [comment]
Ok, forgive me. I was misleaded by my navigation path. 
That code works great. 
The code that is causing me problems is THIS one (that is in the parent page, from which I reach the previous components):

[https://codesandbox.io/s/23lq2o0w7j](https://codesandbox.io/s/23lq2o0w7j)

In particular, the problem seems to be in this part of code: 

```
<GridCell span="4">
          <NavLink to="/destpage">
            <Card>
              <CardPrimaryAction>
              <CardMedia
                square
                style= {{backgroundImage: 'url(https://material-components-web.appspot.com/images/1-1.jpg)'}}
              >
                <CardMediaContent>
                  <Typography>Prova Nav</Typography>
                </CardMediaContent>
              </CardMedia>
            </CardPrimaryAction>
            </Card>
          </NavLink>
       </GridCell> 
``` 
It seems that <CardPrimaryAction> cannot conclude the ripple effect, because the <NavLink> runs the page away before the action is ended.

**Comment by Deilv** [comment]
I have the same issue with totally different code. Any time you destroy the component with a Ripple effect before it completes (ie due to navigation, such as a Login Button) the error appears in the console because instance.root_ is null. I have opened an issue with mdc-web suggesting they add a simple check for sanity. No comment yet.

**Comment by jamesmfriedman** [comment]
Awesome thank you for the additional insight. I’ll find a workaround for rmwc and hopefully it gets fixed up stream.

**Comment by Deilv** [comment]
Here's some additional information about the error, still trying to figure out if this is caused by rmwc.

[Error] TypeError: null is not an object (evaluating 'instance.root_.getBoundingClientRect')
	computeBoundingRect (app.js:3255)
	rmBoundedActivationClasses_ (app.js:3858)
	runDeactivationUXLogicIfReady_ (app.js:3841)
	activationTimerCallback_ (app.js:3505)

**Comment by Deilv** [comment]
Successfully replicated the issue: https://codesandbox.io/s/nk01k2mjjm

**Comment by alecchisi** [comment]
I'll edit the Issue Title

**Comment by jamesmfriedman** [comment]
Thanks @Deilv. The code sandbox replications are hugely helpful. I'll find a workaround.

**Comment by Deilv** [comment]
@jamesmfriedman so far I've traced the problem to the setTimeout in the Foundation class that calls its callback after the component is unmounted, meaning the react ref will return null. Perhaps the solution would be using componentWillUnmount() to clear the timeout, but I got lost between the foundation class and its adapter. My issue report in the mdc-web repository is yet unanswered, maybe you'll have more luck.

**Comment by jamesmfriedman** [comment]
In RMWCFoundation class, there is a setTimeout in componentWillUnmount that keeps the refs around for an extra frame to avoid most errors like this.

I think a simple solution for this is to override componentWillUnmount in the RMWC Ripple Foundation and keep the refs around long enough to avoid this error.

**Comment by jamesmfriedman** [comment]
@Deilv thats such a simple fix... Hope they get it in. I'd rather fix it there then work around it in RMWC.

**Comment by Deilv** [comment]
PR approved, waiting for merge

**Comment by Deilv** [comment]
PR has been merged so the fix should be in the next mdc-web release.

**Comment by jamesmfriedman** [comment]
Great news, they put it in as a patch yesterday so we'll pick it up with the next RMWC release which should be in the next couple of days.

**Comment by alecchisi** [comment]
Thank you!

**Comment by j-o-d-o** [comment]
I am having this error also on checkboxes if they are not initially selected. Good to know this will be fixed in the next release! ; )

**Comment by jamesmfriedman** [comment]
Awesome work on getting this in upstream @Deilv. Fixed with 1.5
